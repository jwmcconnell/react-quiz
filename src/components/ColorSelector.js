import React from 'react';
import PropTypes from 'prop-types';

const ColorSelector = ({ color, onChange }) => {
  return <input type="color" name="color" onChange={onChange} value={color} />;
};

ColorSelector.propTypes = {
  color: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default ColorSelector;
