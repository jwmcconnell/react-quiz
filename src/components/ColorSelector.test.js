import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('ColorSelector component', () => {
  it('renders ColorSelector', () => {
    const wrapper = shallow(<ColorSelector color='#FFF' backgroundColor="#000" onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
