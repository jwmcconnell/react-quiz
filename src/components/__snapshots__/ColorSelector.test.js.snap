// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector component renders ColorSelector 1`] = `
<Fragment>
  <label>
    Text Color:
    <input
      name="color"
      onChange={[Function]}
      type="color"
      value="#FFF"
    />
  </label>
  <label>
    Background Color:
    <input
      name="backgroundColor"
      onChange={[Function]}
      type="color"
      value="#000"
    />
  </label>
</Fragment>
`;
